src/mono/mono/mini/method-to-ir.c

```diff
@@ -4321,15 +4321,16 @@ mini_emit_ldelema_2_ins (MonoCompile *cfg, MonoClass *klass, MonoInst *arr, Mono
 	// FIXME: Do we need to do something here for i8 indexes, like in ldelema_1_ins ?
 #endif
 
+	/* range checking */
 	if (COMPILE_LLVM (cfg)) {
 		// null checking in LLVM to address https://github.com/dotnet/runtime/issues/79022
-		MONO_EMIT_NEW_BIALU_IMM (cfg, OP_COMPARE_IMM, -1, arr->dreg, 0);
-		MONO_EMIT_NEW_COND_EXC (cfg, EQ, "NullReferenceException");
+		MONO_EMIT_NEW_LOAD_MEMBASE_FAULT (cfg, bounds_reg,
```

Comment:
It can use _FAULT in both cases, its the same when not using llvm. _FAULT just means a load which can fault.

Reply:
I restricted it because this only fails on LLVM without the null-check. Other engines do not show the issue and I didn't want to impact the performance there.